src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,166 @@
+    public List<Task> loadTasks() {
+        List<Task> tasks = new ArrayList<>();
+
+        try {
+            File file = new File(filePath);
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+
+            if (file.length() == 0) {
+                // File is empty, return an empty list
+                return tasks;
+            }
+
+            try (Scanner scanner = new Scanner(file)) {
+                while (scanner.hasNextLine()) {
+                    String line = scanner.nextLine();
+//                    System.out.println("Line: " + line); // Add this line for debugging
+                    String[] parts = line.split(" \\| ");
+
+                    if (parts.length >= 3) { // Ensure there are enough parts to proceed
+                        String type = parts[0];
+                        String isDone = parts[1];
+                        String description = parts[2];
+
+                        switch (type) {
+                            case "T":
+                                tasks.add(new ToDo(description, isDone));
+                                break;
+                            case "D":
+                                tasks.add(parseDeadline(parts, isDone));
+                                break;
+                            case "E":
+                                tasks.add(parseEvent(parts, isDone));
+                                break;
+                        }
+                    } else {
+                        System.out.println("Skipping line with insufficient parts: " + line);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            // Handle file reading error
+            e.printStackTrace();
+        }
+
+        return tasks;
+    }
+
+    /**
+     * Parses the saved deadline from the file into the list
+     * @param parts The string array containing the split parts of the task saved into the file
+     * @param isDone Status of whether the deadline is done
+     * @return The deadline task
+     */
+    private Deadline parseDeadline(String[] parts, String isDone) {
+        String description = parts[2];
+        String dueBy = parts[3];
+        LocalDate dueDate = LocalDate.parse(dueBy);  // Parse the date
+        return new Deadline(description, dueDate, isDone);
+    }
+
+
+    /**
+     * Parses the saved event from the file into the list
+     * @param parts The string array containing
+     * @param isDone Status of whether the deadline is done
+     * @return The event task
+     */
+    private Event parseEvent(String[] parts, String isDone) {
+        String description = parts[2];
+        String startTiming = parts[3];
+        String endTiming = parts[4];
+        LocalDate startTime = LocalDate.parse(startTiming);  // Parse the datetime
+        LocalDate endTime = LocalDate.parse(endTiming);  // Parse the datetime
+        return new Event(description, startTime, endTime, isDone);
```

Comment:
I like how you name each variable corresponding to the parts of the string here!